tendrl/monitoring_integration/sync/dashbaord_sync.py

```diff
@@ -47,120 +58,135 @@ def refresh_dashboard(self):
-                                    cluster_details["integration_id"],
-                                    cluster_details["sds_name"]
-                                )
-                    except (KeyError, AttributeError):
-                        alert_utils.delete_alert_dashboard(dashboard_name)
-                        self.create_dashboard(
-                            dashboard_name,
-                            cluster_details
-                        )
-
-                else:
-                    self.create_dashboard(
+                except (KeyError, AttributeError) as ex:
+                    logger.log(
+                        "error",
+                        NS.publisher_id,
+                        {
+                            "message": "Failed to update alert "
+                            "dashboard %s err: %s" % (dashboard_name, ex)
+                        }
+                    )
+                    raise ex
+            else:
+                if cluster_details[dashboard_name]:
+                    resource_json = self.create_dashboard(
                         dashboard_name,
-                        cluster_details
+                        cluster_details[dashboard_name]
                     )
+            if resource_json and not no_changes_in_dashboard:
+                resp = alert_utils.post_dashboard(resource_json)
+                self.log_message(resp, dashboard_name)
 
-    def create_dashboard(self, resource_name, cluster_details):
-        flag = False
-        for resource in cluster_details[resource_name]:
-            if not flag:
-                # create dashboard
-                alert_dashboard.create_resource_dashboard(
-                    resource_name,
-                    resource,
-                    cluster_details["sds_name"],
-                    cluster_details["integration_id"]
-                )
-                flag = True
-            else:
-                # update dashboard
-                self.create_gluster_resource(
-                    [resource],
-                    resource_name,
-                    cluster_details["integration_id"],
-                    cluster_details["sds_name"]
-                )
+    def create_dashboard(self, resource_type, resources):
+        resource_json = {}
+        # create dashboard
+        if resources:
+            # Create operation takes more excution time than update
+            # Create dashboard for one resource using that do update for
+            # other resources
+            resource = resources.pop()
+            resource_json = alert_dashboard.create_resource_dashboard(
+                resource_type,
+                resource
+            )
+        # After creating first panel update the other panels
+        if resources and resource_json:
+            # update dashboard
+            rows = resource_json["dashboard"]["rows"]
+            highest_panel_id = rows[-1]["panels"][-1]["id"]
```

Comment:
I understand this is `most_recent_panel_id`. Name it accordingly

Reply:
i will rename this to most_recent_panel_id